cloud/docker_client.go

```diff
@@ -427,8 +427,7 @@ func (c *dockerClientImpl) GetDockerStatus(ctx context.Context, containerID stri
 	}
 	container, err := c.GetContainer(ctx, parent, containerID)
 	if err != nil {
-		if strings.Contains(err.Error(), "No such container") ||
-			strings.Contains(err.Error(), "Is the docker daemon running?") {
+		if docker.IsErrNotFound(err) || docker.IsErrConnectionFailed(err) {
```

Comment:
In 2 functions in docker.go we return StatusTerminated and "error making connection" for client.IsErrConnectionFailed. I don't think `HasStarted: false` would be a certainty is the container is terminated or if we simply can't connect to it.